### PR TITLE
chore: Make 0004 required for current posthog version as the max version

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

There are already problems, when this migration isn't completed, so let's make sure folks complete it now if not earlier.
More context: https://posthog.slack.com/archives/C0374DA782U/p1655132304877339

## Changes

This PR makes the [0004 async migration](https://github.com/PostHog/posthog/blob/7ac6efd839d507a5c45348d559db18b7976f6c62/posthog/async_migrations/migrations/0004_replicated_schema.py) for self-hosted instances upgrades. If you haven't already ran the async migration, make sure to run it before attempting a helm upgrade. 

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by macobo** [comment]
This requires further discussion - let's not yolo-merge this. We set the contract we'll let people know at least one release.

**Comment by tiina303** [comment]
So the options I see:
1. we merge this, so going forward when upgrading people don't run into the problems like broken ingestion due to this
2. we investigate and figure out when and what became a problem with the 0004 migration and fix it making things work without having ran 0004 now, then make it required later in the future.

IMHO (2) is a waste of time and I'd prefer to move fast here as the end state will anyway be that 0004 is required, likely many people have already ran the migration too.

File: posthog/async_migrations/migrations/0004_replicated_schema.py [hunk 1]
```diff
@@ -100,6 +100,8 @@ class Migration(AsyncMigrationDefinition):
 
     depends_on = "0003_fill_person_distinct_id2"
 
+    posthog_max_version = "1.36.1"
```
Comment (yakkomajuri) [inline posthog/async_migrations/migrations/0004_replicated_schema.py:103]: let's maintain convention of having this follow minor releases

```suggestion
    posthog_max_version = "1.36.99"
```

**Review by yakkomajuri — APPROVED** [review]
I'm happy with this (if my suggestion goes in).

Can you just take a look at the failing test though?

**Comment by macobo** [comment]
If we end up merging this - please also add a highlight tag to help your friendly release notes writer. Even better, why not write the release notes under this PR?

**Pull request merged**